tree/ntuplebrowse/v7/test/ntuple_browse.cxx

```diff
@@ -0,0 +1,500 @@
+using RNTupleReader = ROOT::Experimental::RNTupleReader;
+using RNTupleWriter = ROOT::Experimental::RNTupleWriter;
+
+/* This test cannot test the entire TBrwoser-support, it has no access to the TBrowser GUI.
+ * It only test certain functions, things it cannot test are:
+ * - The way the ntuple is displayed in TBrowser.
+ * - If histogram is displayed correctly.
+ * - It doesn't know what happens after TBrowser::Add() is called, and also doesn't know how often it was called.
+ */
+
+namespace {
+  /**
+   * An RAII wrapper around an open temporary file on disk. It cleans up the guarded file when the wrapper object
+   * goes out of scope.
+   */
+class FileRaii {
+private:
+   std::string fPath;
+   TFile *file;
+public:
+   FileRaii(const std::string &path) :
+      fPath(path),
+      file(TFile::Open(fPath.c_str(), "RECREATE"))
+      { }
+   FileRaii(const FileRaii&) = delete;
+   FileRaii& operator=(const FileRaii&) = delete;
+   TFile* GetFile() const { return file; }
+   ~FileRaii() {
+      file->Close();
+      std::remove(fPath.c_str());
+   }
+};
+} // anonymous namespace
+
+// helper function for tests
+std::string GetFileName(const TDirectory *directory) {
+   std::string fullPath(directory->GetPath());
+   return std::string(fullPath, 0, fullPath.find(".root") + 5);
+}
+
+TEST(RNTupleBrowse, Floattest)
+{
+   FileRaii fileGuard("test.root");
+   TDirectory *directory = fileGuard.GetFile()->mkdir("Staff");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldPt = model->MakeField<float>("FloatField");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), directory->GetName(), GetFileName(directory));
+      *fieldPt = 5.0f;
+      ntuple->Fill();
+   } // flushes content to test.root
+   
+   RNTupleBrowser browseObject(directory);
+   browseObject.SetDirectory(directory);
+   browseObject.Browse(nullptr);
+   
+   ASSERT_EQ(browseObject.fNTupleBrowsePtrVec.size(), static_cast<std::size_t>(1));
+   ASSERT_STREQ(static_cast<RNonFolder*>(browseObject.fNTupleBrowsePtrVec.at(0).get())->GetName(), "FloatField");
+}
+
+
+
+TEST(RNTupleBrowse, Stringtest)
+{
+   
+   FileRaii fileGuard("test2.root");
+   TDirectory *directory = fileGuard.GetFile()->mkdir("Staff2");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldPt = model->MakeField<std::string>("StringField");
+      auto fieldPt2 = model->MakeField<std::string>("StringField2");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), directory->GetName(), GetFileName(directory));
+      *fieldPt = "TestString";
+      ntuple->Fill();
+   } // flushes content to test.root
+   
+   RNTupleBrowser browseObject(directory);
+   browseObject.SetDirectory(directory);
+   browseObject.Browse(nullptr);
+   
+   ASSERT_EQ(browseObject.fNTupleBrowsePtrVec.size(), static_cast<std::size_t>(2));
+   ASSERT_STREQ(static_cast<RNonFolder*>(browseObject.fNTupleBrowsePtrVec.at(0).get())->GetName(), "StringField");
```

Comment:
We should distinguish between `ASSERT_...` and `EXPECT_...` macros.  An `ASSERT_...` stops the entire test case because if it fails nothing afterwards can be meaningful.  The `EXPECT_...` macros simply report an error and move on.  In this case, for instance, the first macro is an ASSERT, the second and third should be EXPECT.